cli/tests/unit/net_test.ts

```diff
@@ -385,6 +385,70 @@ Deno.test(
   },
 );
 
+Deno.test(
+  { permissions: { net: true } },
+  async function netUdpSendReceiveTestSizeLimits() {
+    // Ensure payload being sent is within UDP limit, which seems to be 65507
+    // bytes
+    const alice = Deno.listenDatagram({
+      port: listenPort,
+      transport: "udp",
+      hostname: "127.0.0.1",
+    });
+    // wrap this in a try/catch so other tests can continue if we fail
+    // if we don't close here then listening on future tests fails
+    try {
+      assert(alice.addr.transport === "udp");
+      assertEquals(alice.addr.port, listenPort);
+      assertEquals(alice.addr.hostname, "127.0.0.1");
+    } catch (err) {
+      alice.close();
+      throw err;
+    }
+
+    const bob = Deno.listenDatagram({
+      port: listenPort2,
+      transport: "udp",
+      hostname: "127.0.0.1",
+    });
+    try {
+      assert(bob.addr.transport === "udp");
+      assertEquals(bob.addr.port, listenPort2);
+      assertEquals(bob.addr.hostname, "127.0.0.1");
+    } catch (err) {
+      bob.close();
+      throw err;
+    }
+
+    const sizes = [0, 1, 2, 256, 1024, 4096, 16384, 65506, 65507, 65508, 65536];
+    const rx = /Message too long \(os error \d\d\)/;
+
+    for (const size of sizes) {
+      const tosend = new Uint8Array(size);
+      let byteLength = 0;
+      try {
+        byteLength = await alice.send(tosend, bob.addr);
+      } catch (err) {
+        // Note: we have to do the test this way as different OS's have
+        // different UDP size limits enabled, so we will just ensure if
+        // an error is thrown it is the one we are expecting.
+        assert(err.message.match(rx));
```

Comment:
This test is failing on Windows on this line. cc @billywhizz

Reply:
yeah. every OS has a different error message seemingly. i'll see if i can fix for windows.

Reply:
i fixed this but now seeing a failure here on linux which wasn't happening before. are these just flaky tests?

https://github.com/denoland/deno/actions/runs/7357751549/job/20029885353?pr=21611#step:42:3062

Reply:
I'm seeing that in another PR too. Looks like a flake yeah

Reply:
i'm seeing a SERVFAIL status when i try to resolve that address locally using dig. so, possibly it is an intermittent issue on google dns side?

```shell
dig 8.8.8.8.in-addr.arpa

; <<>> DiG 9.18.18-0ubuntu0.22.04.1-Ubuntu <<>> 8.8.8.8.in-addr.arpa
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: SERVFAIL, id: 7815
;; flags: qr rd ra; QUERY: 1, ANSWER: 0, AUTHORITY: 0, ADDITIONAL: 1

;; OPT PSEUDOSECTION:
; EDNS: version: 0, flags:; udp: 65494
;; QUESTION SECTION:
;8.8.8.8.in-addr.arpa.          IN      A

;; Query time: 196 msec
;; SERVER: 127.0.0.53#53(127.0.0.53) (UDP)
;; WHEN: Fri Dec 29 16:12:29 GMT 2023
;; MSG SIZE  rcvd: 49
```

i also see this when i query google 8.8.8.8 directly. :thinking: 

```shell
dig @8.8.8.8 8.8.8.8.in-addr.arpa

; <<>> DiG 9.18.18-0ubuntu0.22.04.1-Ubuntu <<>> @8.8.8.8 8.8.8.8.in-addr.arpa
; (1 server found)
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: SERVFAIL, id: 35308
;; flags: qr rd ra; QUERY: 1, ANSWER: 0, AUTHORITY: 0, ADDITIONAL: 1

;; OPT PSEUDOSECTION:
; EDNS: version: 0, flags:; udp: 512
; EDE: 0 (Other): ([209.244.0.1] Lame delegation at 8.8.8.in-addr.arpa for 8.8.8.8.in-addr.arpa/a)
; EDE: 0 (Other): ([209.244.0.2] Lame delegation at 8.8.8.in-addr.arpa for 8.8.8.8.in-addr.arpa/a)
; EDE: 22 (No Reachable Authority): (At delegation 8.8.8.in-addr.arpa for 8.8.8.8.in-addr.arpa/a)
;; QUESTION SECTION:
;8.8.8.8.in-addr.arpa.          IN      A

;; Query time: 56 msec
;; SERVER: 8.8.8.8#53(8.8.8.8) (UDP)
;; WHEN: Fri Dec 29 16:16:11 GMT 2023
;; MSG SIZE  rcvd: 282
```

Reply:
i also see the same thing here. sounds like something might be temporarily broken on google side?

https://dns.google/query?name=8.8.8.8.in-addr.arpa